Allow for Custom Runtime Identifiers

atlasapplications opened the pull request:
The C# export process assigns a runtime identifier based off predefined options depending on the exporting platform and architecture. This makes sense almost all the time but I found a case where this creates a limitation. NativeAOT for android requires the linux-bionic-arm64 runtime identifier which is not what Godot assigns (it uses android-arm64). This can be solved by simply creating an override in the export settings for advanced users that want to try a custom runtime identifier. The default functionality is untouched if the option is left blank.

Resolves #97775

Calinou committed:
Disable unused Basis Universal features to reduce binary size

aaronfranke committed:
Add a sort method to Dictionary and HashMap

reduz committed:
Make RID_Owner lock-free for fetching.

This PR makes RID_Owner lock free for fetching values, this should give a very
significant peformance boost where used.

Some considerations:

* A maximum number of elements to alocate must be given (by default 256k).
* Access to the RID structure is still safe given they are independent from addition/removals.
* RID access was never really thread-safe in the sense that the contents of the data are not protected anyway. Each server needs to implement locking as it sees fit.

ryevdokimov committed:
Offset drag instantiated scenes that result in a collision by its bounds to prevent overlap

Co-Authored-By: Robbie Cooper <cooperra@users.noreply.github.com>

vPumpking committed:
Add filename filter field to FileDialog

Added possibility to filter files and folders through a text entry.

shatyuka committed:
Apply `WS_MINIMIZE` style on window creation

havi05 committed:
Add `Auto width` behavior to ItemList

Co-authored-by: Craig Hupin <althar93@hotmail.com>

KoBeWi committed:
Call restart_editor() in RUN_PROJECT_MANAGER

KoBeWi committed:
Close expanded tile editor when inspector updates

KoBeWi committed:
Don't fold resources when child of main inspector exits

KoBeWi committed:
Misc code cleanup in EditorFileDialog

KoBeWi committed:
Allow horizontal scrolling in Tree using Shift

timothyqiu committed:
Move pseudolocalization into TranslationDomain

Also adds command-line option `--editor-pseudolocalization`

KoBeWi committed:
Add _resource_changed() helper method to EditorResourcePicker

KoBeWi committed:
Globally remember advanced toggle in project settings

bruvzg committed:
[Editor] Try loading system Noto CJK font.

dsnopek committed:
GDExtension: Allow directly getting `ObjectID` from `Variant`

D0V4HKIIN committed:
Add unit tests for PhysicsMaterial

KoBeWi committed:
Print error on invalid call_group() calls

kitbdev committed:
Fix TabBar initialization issue and add tests

KoBeWi committed:
Fix ScrollContainer configuration warnings

KoBeWi committed:
Add submenu support to EditorContextMenuPlugin

kroketio committed:
Metal: implement texture_create_from_extension

Parameters p_type, p_format, p_array_layers, p_depth_stencil are
ignored - MTLTexture (and the callee) already have this information
and is only relevant when reinterpreting or remaping the texture in
different ways.

reduz committed:
Make internal unique scene resource ID deterministic

Changes the Resource::generate_scene_unique_id() to be deterministic and
seedable.

Fixes #97110

rsubtil committed:
Support object inspection through DAP `variables` request

bruvzg committed:
[Docs] Adds notes about glyph contour decomposition format.

KoBeWi committed:
Fix closing Theme Editor not actually closing it

Commit:
Use dedicated `print_error` method for colored output for unsupported drivers

Signed-off-by: Yevhen Babiichuk (DustDFG) <dfgdust@gmail.com>

Commit:
Replace comments with printed warning for Metal on x86_64

Signed-off-by: Yevhen Babiichuk (DustDFG) <dfgdust@gmail.com>

timothyqiu committed:
Fix `atr_n()` behavior when auto translation is disabled

ydeltastar committed:
Make the "Quick Open" dialog available via EditorInterface

rsubtil committed:
Support REPL expressions through DAP `evaluate` request

bruvzg committed:
[macOS] Fix menu crash when used from opened native dialog.

bruvzg committed:
[Linux/BSD] Fix cross-device rename.

bruvzg committed:
[DisplayServer] Implement has_hardware_keyboard method for Android and iOS.

adamscott committed:
Fix UI of sun angle in the sun/preview env popup

bruvzg committed:
Fix window_set_current_screen for maximized windows.

dalexeev committed:
GDScript: Fix `GDScriptCache::get_full_script()` uses non-remapped path

atlasapplications committed:
Allow for Custom Runtime Identifiers

The C# export process assigns a runtime identifier based off predefined options depending on the exporting platform and architecture. This makes sense almost all the time but I found a case where this creates a limitation. NativeAOT for android requires the linux-bionic-arm64 runtime identifier which is not what Godot assigns (it uses android-arm64). This can be solved by simply creating an override in the export settings for advanced users that want to try a custom runtime identifier. The default functionality is untouched if the option is left blank.

atlasapplications committed:
Allow for Custom Runtime Identifiers

The C# export process assigns a runtime identifier based off predefined options depending on the exporting platform and architecture. This makes sense almost all the time but I found a case where this creates a limitation. NativeAOT for android requires the linux-bionic-arm64 runtime identifier which is not what Godot assigns (it uses android-arm64). This can be solved by simply creating an override in the export settings for advanced users that want to try a custom runtime identifier. The default functionality is untouched if the option is left blank.

raulsntos commented:
I don't think you should be able to override the runtime identifier. When exporting, Godot will use the template for the target platform and architectures, if you override the runtime identifier you may not match the template. For example, if you are exporting to Android but override the runtime identifier with `win-x64`.

And specifically about Android, since the runtime identifier also contains the architecture, if you override it with `linux-bionic-arm64` as you mentioned. What if you had other ABIs enabled? Like `x86_32` and `arm32`. Godot will include those in the APK but they will not work, because you are building the C# project with the runtime identifier `linux-bionic-arm64` for all the ABIs.

I would prefer a boolean option only available to Android exports to enable AOT which would automatically pick the correct runtime identifier (i.e.: pick `linux-bionic` when true, and `android` when false which should be the default).

bruvzg committed:
[iOS] Add support for dark and tinted icon versions.

rsubtil committed:
Bump DAP requests timeout to 5 seconds

clayjohn committed:
Properly utilize surface information when creating pipelines and vertex arrays in canvas renderer

akien-mga committed:
Merge pull request #97792 from clayjohn/RD-canvas-vertex-format

Properly utilize surface information when creating pipelines and vertex arrays in canvas renderer

akien-mga committed:
Merge pull request #77213 from aaronfranke/sort-dict

Add a sort method to Dictionary and HashMap

akien-mga committed:
Merge pull request #88673 from vPumpking/FileDialog-Filter

Add filename filter field to FileDialog

akien-mga committed:
Merge pull request #93270 from Gamemap/feature-itemlist-autowidth

Add `Auto width` behavior to ItemList

akien-mga committed:
Merge pull request #96230 from timothyqiu/per-domain-pseudolocalization

Move pseudolocalization into `TranslationDomain`

akien-mga committed:
Merge pull request #96850 from bruvzg/noto_cjk_lookup

[Editor] Try loading system Noto CJK font.

akien-mga committed:
Merge pull request #97227 from D0V4HKIIN/master

Add unit tests for PhysicsMaterial

akien-mga committed:
Merge pull request #97255 from kitbdev/test-tabs

Fix TabBar initialization issue and add tests

akien-mga committed:
Merge pull request #97465 from DarioSamo/lock-free-rid

Make `RID_Owner` lock-free for fetching.

akien-mga committed:
Merge pull request #97514 from bruvzg/font_contour_info

[Docs] Add notes about glyph contour decomposition format.

akien-mga committed:
Merge pull request #97585 from rsubtil/feature-dap_object_inspection

Support object inspection through DAP

akien-mga committed:
Merge pull request #97624 from timothyqiu/atrn

Fix `atr_n()` behavior when auto translation is disabled

akien-mga committed:
Merge pull request #97678 from bruvzg/fix_nd_menu

[macOS] Fix menu crash when used from opened native dialog.

akien-mga committed:
Merge pull request #97719 from bruvzg/cross_dev_rename

[Linux/BSD] Fix cross-device rename.

akien-mga committed:
Merge pull request #97743 from bruvzg/has_kbd

[DisplayServer] Implement `has_hardware_keyboard` method for Android and iOS.

akien-mga committed:
Merge pull request #97747 from adamscott/ux-sun-angle

Improve UI of sun angle in the sun/preview env popup

akien-mga committed:
Merge pull request #97757 from bruvzg/max_move_screen

Fix `window_set_current_screen` for maximized windows.

akien-mga committed:
Merge pull request #97769 from dalexeev/gds-fix-gdscript-cache-path

GDScript: Fix `GDScriptCache::get_full_script()` uses non-remapped path

WhalesState committed:
Expose `LineEdit` `edit` and `unedit` methods.

akien-mga committed:
Merge pull request #85321 from Calinou/basis-universal-disable-unused-features

Disable unused Basis Universal features to reduce binary size

akien-mga committed:
Merge pull request #91487 from shatyuka/window-style

Apply `WS_MINIMIZE` style on window creation

akien-mga committed:
Merge pull request #97304 from kroketio/metal-return-logical-device-handle

Metal: Implement `texture_create_from_extension`

Geometror committed:
[GraphEdit] Only print warning for connection layer deletion when justified

DarioSamo committed:
Configure MSAA properly in canvas renderer's pipelines.

akien-mga committed:
Merge pull request #97094 from WhalesState/expose-edit-unedit

Expose `LineEdit` `edit` and `unedit` methods.

akien-mga committed:
Merge pull request #97119 from dsnopek/gdextension-object-instance-id

GDExtension: Allow directly getting `ObjectID` from `Variant`

akien-mga committed:
Merge pull request #97362 from reduz/make-unique-scene-id-deterministic

Make internal unique scene resource ID deterministic

akien-mga committed:
Merge pull request #97552 from dustdfg/drivers/use_dedicated_print_error

Use dedicated `print_error` method for colored output for unsupported drivers

akien-mga committed:
Merge pull request #97633 from ydeltastar/quickly-quick-open

Make the "Quick Open" dialog available via `EditorInterface`

akien-mga committed:
Merge pull request #97789 from bruvzg/ios_icons

[iOS] Add support for dark and tinted icon versions.

bruvzg committed:
Partially revert 96780, remove warnings from project/editor settings `_get`.

passivestar committed:
Fix quick open background panel style

akien-mga committed:
Merge pull request #88511 from ryevdokimov/offset-drag-to-instantiate-scene

Offset drag instantiated scenes that result in a collision by its bounds to mitigate overlap

akien-mga committed:
Merge pull request #94880 from KoBeWi/never-before-seen_way_to_restart_editor

Call `restart_editor()` in `RUN_PROJECT_MANAGER`

akien-mga committed:
Merge pull request #95034 from KoBeWi/take_a_problem_and_make_it_worse._solved

Close expanded tile editor when inspector updates

akien-mga committed:
Merge pull request #95149 from KoBeWi/inspector_mains

Don't fold resources when child of main inspector exits

akien-mga committed:
Merge pull request #95544 from KoBeWi/clean_code™

Misc code cleanup in EditorFileDialog

akien-mga committed:
Merge pull request #95889 from KoBeWi/tree_growing_sideways

Allow horizontal scrolling in Tree using Shift

akien-mga committed:
Merge pull request #96379 from KoBeWi/wind_of_resource_changes

Add `_resource_changed()` helper method to EditorResourcePicker

akien-mga committed:
Merge pull request #96615 from KoBeWi/run_as_administrator

Globally remember advanced toggle in project settings

akien-mga committed:
Merge pull request #97240 from KoBeWi/wrong_number

Print error on invalid `call_group()` calls

akien-mga committed:
Merge pull request #97291 from KoBeWi/visible_in_tree_was_a_billion_dollar_mistake

Fix ScrollContainer configuration warnings

akien-mga committed:
Merge pull request #97292 from KoBeWi/infinite_submenus_in_your_editor

Add submenu support to EditorContextMenuPlugin

akien-mga committed:
Merge pull request #97543 from KoBeWi/to_edit_or_not_to_edit

Fix closing Theme Editor not actually closing it

akien-mga committed:
Merge pull request #97570 from dustdfg/ios_macos/print_warning_instead_of_comments

Replace comments with printed warning for Metal on x86_64

akien-mga committed:
Merge pull request #97809 from Geometror/fix-ge-conn-layer-warning

[GraphEdit] Only print warning for connection layer deletion when justified

akien-mga committed:
Merge pull request #97811 from DarioSamo/2d-msaa-fix

Configure MSAA properly in canvas renderer's pipelines.

akien-mga committed:
Merge pull request #97817 from bruvzg/prop_warn_remove

Partially revert #96780, remove warnings from project/editor settings `_get`.

akien-mga committed:
Merge pull request #97820 from passivestar/quick-open-bg

Fix quick open background panel style

atlasapplications committed:
Create Android NativeAOT Export Template

This allows for Godot to publish C# enabled projects as NativeAOT for Android. This also reverses the ability to use custom runtime identifiers.

atlasapplications committed:
Expand NativeAOT templates to other platforms

Previously, the templates only would've worked on Android, but now other supported platforms are included. Disabled visibility of option for iOS. Made visibility conditional as an advanced option.

atlasapplications committed:
Merge branch 'custom-runtimeidentifier' of https://github.com/atlasapplications/godot into custom-runtimeidentifier
